### PR TITLE
Validate inputs for `searchSubmission` filters and orderBy fields

### PR DESCRIPTION
**Pull request by evan-desu**
Part of #257 

# What changed
1. Added a helper function `validateSubmissionSearchFilters` that:
- Prevents ordering by `id`.
- Prevents ordering by fields that do not exist.
- Limits the number of `fieldsToOrder` to a maximum of two fields.
- Places a minimum of 0 and a maximum of 1000 in the `limit` field (0 returns the default limit of 20).
- Requires at least one `spokenLanguage` property to be filled in.

# Testing instructions
1. Run this query in GraphQl Studio:
```
query GetSubmissions {
    submissions(filters: { 
        isUnderReview: false,
        limit: 5,
        orderBy: [
            {
                fieldToOrder: "healthcareProfessionalName", 
                orderDirection: desc 
            }
        ]
        }) {
        id
        googleMapsUrl
        healthcareProfessionalName
        isUnderReview
        isApproved
        isRejected
        createdDate
        updatedDate
        spokenLanguages {
            iso639_3
            nameEn
            nameJa
            nameNative
        }
    }
}
```
It should succeed.
2. Change `limit: 5` to `limit: -1` or `limit: 1001` and it should fail.
3. Add the filter `healthcareProfessionalName: " "` and it should fail.

### CODE REVIEW

File: src/services/submissionService.ts [hunk 1]
```diff
@@ -56,8 +56,118 @@ function validateIdInput(id: string): Result<gqlTypes.Submission> {
...
+    if (limit && (limit > 1000)) {
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'limit',
+            errorCode: ErrorCode.MAX_LIMIT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.googleMapsUrl === '' || (filters.googleMapsUrl && !filters.googleMapsUrl.trim())) {
```
Comment (ermish) [inline src/services/submissionService.ts:124]:
```suggestion
    if (!filters.googleMapsUrl || !filters.googleMapsUrl.trim()) {
```

I think this is equivalent 🤔

File: src/services/submissionService.ts [hunk 2]
```diff
@@ -56,8 +56,118 @@ function validateIdInput(id: string): Result<gqlTypes.Submission> {
...
+        validationResults.hasErrors = true
+        validationResults.errors?.push({
+            field: 'googleMapsUrl',
+            errorCode: ErrorCode.MISSING_INPUT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.healthcareProfessionalName === '' ||
```
Comment (ermish) [inline src/services/submissionService.ts:133]: same here

File: src/services/submissionService.ts [hunk 3]
```diff
@@ -56,8 +56,118 @@ function validateIdInput(id: string): Result<gqlTypes.Submission> {
...
+        validationResults.errors?.push({
+            field: 'healthcareProfessionalName',
+            errorCode: ErrorCode.MISSING_INPUT,
+            httpStatus: 400
+        })
+    }
+
+    if (filters.spokenLanguages) {
+        for (const lang of filters.spokenLanguages) {
+            if ((lang.iso639_3 === '' || (lang.iso639_3 && !lang.iso639_3.trim())) &&
```
Comment (ermish) [inline src/services/submissionService.ts:145]: same for these

**Review by ermish — CHANGES_REQUESTED** [review]
nice! looks great. I added a small comment

**Comment by theyokohamalife** [comment]
@evan-desu  Can you write a descriptive PR title?

File: src/services/submissionService.ts (hunk 1 above)
Comment (evan-desu) [reply src/services/submissionService.ts:124]: @ermish I tried this and found it throws an error if `googleMapsUrl` is falsy, meaning it’ll thrown an error when it’s not provided as a search filter. The original throws an error only if `googleMapsUrl` is chosen as a search filter but contains an empty string or whitespace that becomes empty after trimming. How do we want to handle this?

File: src/services/submissionService.ts (hunk 2 above)
Comment (evan-desu) [reply src/services/submissionService.ts:133]: @ermish See the above comment regarding `googleMapsUrl`.

**Comment by evan-desu** [comment]
> @evan-desu Can you write a descriptive PR title?

@theyokohamalife Updated! Let me know if it's okay.

**Comment by theyokohamalife** [comment]
Closing this since major changes will be merged in https://github.com/ourjapanlife/findadoc-server/pull/325

**Pull request closed without merging**